contrib/devtools/update-translations.py

```diff
@@ -172,27 +182,60 @@ def postprocess_translations(reduce_diff_hacks=False):
 
                     for error in errors:
                         print('%s: %s' % (filename, error))
-
-                    if not valid: # set type to unfinished and clear string if invalid
-                        translation_node.clear()
-                        translation_node.set('type', 'unfinished')
-                        have_errors = True
-
+						
+                    # check if translation can be fixed
+                    if not valid:
+                       if translation_node.text != None: # only attempt to fix if translation is not a NoneType object
+                           translation_node.text = fix_string(translation_node.text) #fix most common mistakes by replacing symbols
+                           translation_f = split_format_specifiers(find_format_specifiers(translation_node.text))
+                           if source_f == translation_f: # check if translation is acceptable after fixing it. if not, try to fix more.
+                               if translation_node.text[0] != "%" and translation_node.text.find(' %') == -1 and translation_node.text.find('(%') == -1 and translation_node.text.find('\'%') == -1 and translation_node.text.find('\"%') == -1: #if the translation seems okay, add spaces before % if needed - only if certain strings are not found and if '%' is not the first symbol in a string.
```

Comment:
Instead of adding a very long comment at the end of an already long line of code, you could you bring the it back over above the this line and break it up over multiple lines if needed.